docs/working-with-streams.md

```diff
@@ -1,27 +1,47 @@
 # Streams
 
-## All data live in streams
+Timeplus `streams` are conceptually similar to `tables` in traditional SQL databases — they both hold data. However, there are key differences:
 
-Timeplus is a streaming analytics platform and data lives in streams. Timeplus `streams` are similar to `tables` in the traditional SQL databases. Both of them are essentially datasets. The key difference is that Timeplus stream is an append-only (by default), unbounded, constantly changing events group.
+* A **Timeplus stream** tracks changes and updates through its underlying **Write-Ahead Log (WAL)**, which powers incremental processing.
+* Timeplus supports **both incremental stream processing and historical queries** over stream data.
 
-Timeplus supports multiple types of streams:
+## Types of Streams
 
-1. By default, the streams are append-only and immutable (older data can be purged automatically by setting a retention policy).
-2. If you want to create a stream to track the latest value for a primary key or a set of keys, you can create [Mutable Streams](/mutable-stream). This is only available in Timeplus Enterprise.
-3. In [Timeplus Proton](/proton), you can also create [Versioned Streams](/versioned-stream) and [Changelog Stream](/changelog-stream). But those 2 stream modes will be deprecated and replaced by mutable streams.
-4. You can also define [External Streams](/external-stream) to run SQL against remote Kafka/Redpanda brokers, or the other Timeplus/Proton server.
+To support a variety of use cases efficiently, Timeplus offers multiple types of streams:
 
-## Create a stream
-You can create a stream via the Timeplus Console UI, or via [SQL](/sql-create-stream). When you [ingest data](/ingestion) into Timeplus from Kafka or file sources, streams can be created automatically to match the schema of the data.
+1. **Append Stream**
```

Comment:
For those 5 stream types, maybe you can add markdown link to those 5 subpages, so that they can click and see more details